src/tokens/aoTokens/ao.ts

```diff
@@ -225,6 +225,10 @@ export async function getAoTokenBalance(
 
   const aoToken = aoTokens.find((token) => token.processId === process);
 
+  if (!aoToken) {
+    throw new Error(`Token not found for address '${address}'.`);
```

Comment:
I think this could lead to confusion when a user might remove a token from the cached token list and when the dev asks for the balance it would throw an error, even if the user has balance on that token

Reply:
@7i7o I think we are using the cached token list for both `userTokens` and `tokenBalance`. In case of `tokenBalance`, if `aoToken` is undefined should we fetch the token info and then fetch and return the balance instead of throwing the error ?

Reply:
I think we should just try to fetch the balance and just raise the error if it throws 🤷 
I wouldn't add the extra Info check, because that info isn't returned, we're adding an extra delay and the dev is passing the actual process Id. If they are not using one of the user's cached tokens, they must check the token process id themselves

Reply:
So if `aoToken` is `undefined` should we just return the value we get from the process because for the tokens that are present on the cache we do the following to return the balance ?

```typescript
return new Quantity(BigInt(balance), BigInt(aoToken.Denomination));
```

Reply:
Yes, if `aoToken` is `undefined` we should dryRun and return according to the dryRun result:
- dryRun with a balance msg:  `new Quantity(BigInt(balance), BigInt(aoToken.Denomination));`
- dryRun with no messages (means no balance handler): `throw Error`
- dryRun with errors (means maybe the process was not found): `throw Error`

Reply:
No. You are right. We could fetch `Info` to be consistent 👍
Maybe fetch `Info` only after the balance returns a `>0` amount. Wdyt?